homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -118,15 +155,41 @@ async def _async_step_device_config_if_needed(self) -> FlowResult:
 
         self.context["title_placeholders"] = {"name": name}
 
-        unique_id = await afsapi.get_radio_id()
-        await self.async_set_unique_id(unique_id)
-        self._abort_if_unique_id_configured()
+        # unique_id will already be set when discovered through SSDP with the SSDP UDN,
+        # however, when adding a device manually, it will still be empty at this point.
+        # Now we have successfully logged in, we can use the radio_id of this device as
+        # an unique_id
+        if self.unique_id is None:
+            unique_id = await afsapi.get_radio_id()
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+
+        if show_confirm:
+            self._name = name
+            return await self.async_step_confirm()
 
         return self.async_create_entry(
             title=name,
             data={CONF_WEBFSAPI_URL: self._webfsapi_url, CONF_PIN: DEFAULT_PIN},
         )
 
+    async def async_step_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Allow the user to confirm adding the device. Used when the default PIN could successfully be used."""
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self._name,
+                data={
+                    CONF_WEBFSAPI_URL: self._webfsapi_url,
+                    CONF_PIN: DEFAULT_PIN,
+                },
+            )
```

Comment:
This could be a bit more DRY if you move this into an `_async_create_entry` function since its nearly the same code at the end of `_async_step_device_config_if_needed`

Reply:
I've now re-added this function. I had to remove it during the review of the previous PR.